src/inmanta/deploy/scheduler.py

```diff
@@ -36,11 +39,63 @@
 LOGGER = logging.getLogger(__name__)
 
 
-# FIXME[#8008] review code structure + functionality + add docstrings
-# FIXME[#8008] add import entry point test case
+class TaskManager(abc.ABC):
```

Comment:
I added this to have a restricted interface with the tasks, so they don't need to access private attributes. As I see it, this makes it easier to reason on scheduler state, because it's all in one place (here), and we know it won't be misused elsewhere. The two modules / classes are still relatively tightly coupled, but at least in a well defined manner.